docs/app/routes/docs/guides/react-three-fiber.mdx

```diff
@@ -0,0 +1,401 @@
+import { useSpring, animated } from '@react-spring/three'
+import { Canvas, useThree } from '@react-three/fiber'
+import { MeshDistortMaterial } from '@react-three/drei'
+
+const AnimatedMeshDistortMaterial = animated(MeshDistortMaterial)
+
+const MyScene = () => {
+  const isOver = useRef(false)
+
+  const { width, height } = useThree(state => state.size)
+
+  const [springs, api] = useSpring(
+    () => ({
+      scale: 1,
+      position: [0, 0],
+      color: '#ff6d6d',
+      config: key => {
+        switch (key) {
+          case 'scale':
+            return {
+              mass: 4,
+              friction: 10,
+            }
+          case 'position':
+            return { mass: 4, friction: 220 }
+          default:
+            return {}
+        }
+      },
+    }),
+    []
+  )
+
+  const handleClick = useCallback(() => {
+    let clicked = false
+
+    return () => {
+      clicked = !clicked
+      api.start({
+        color: clicked ? '#569AFF' : '#ff6d6d',
+      })
+    }
+  }, [])
+
+  const handlePointerEnter = () => {
+    api.start({
+      scale: 1.5,
+    })
+  }
+
+  const handlePointerLeave = () => {
+    api.start({
+      scale: 1,
+    })
+  }
+
+  const handleWindowPointerOver = useCallback(() => {
+    isOver.current = true
+  }, [])
+
+  const handleWindowPointerOut = useCallback(() => {
+    isOver.current = false
+
+    api.start({
+      position: [0, 0],
+    })
+  }, [])
+
+  const handlePointerMove = useCallback(
+    e => {
+      if (isOver.current) {
+        const x = (e.offsetX / width) * 2 - 1
+        const y = (e.offsetY / height) * -2 + 1
+
+        api.start({
+          position: [x * 5, y * 2],
+        })
+      }
+    },
+    [api, width, height]
+  )
+
+  useEffect(() => {
+    window.addEventListener('pointerover', handleWindowPointerOver)
+    window.addEventListener('pointerout', handleWindowPointerOut)
+    window.addEventListener('pointermove', handlePointerMove)
+
+    return () => {
+      window.removeEventListener('pointerover', handleWindowPointerOver)
+      window.removeEventListener('pointerout', handleWindowPointerOut)
+      window.removeEventListener('pointermove', handlePointerMove)
+    }
+  }, [handleWindowPointerOver, handleWindowPointerOut, handlePointerMove])
+
+  return (
+    <animated.mesh
+      onPointerEnter={handlePointerEnter}
+      onPointerLeave={handlePointerLeave}
+      onClick={handleClick()}
+      scale={springs.scale}
+      position={springs.position.to((x, y) => [x, y, 0])}>
+      <sphereGeometry args={[1.5, 64, 32]} />
+      <AnimatedMeshDistortMaterial
+        speed={5}
+        distort={0.5}
+        color={springs.color}
+      />
+    </animated.mesh>
+  )
+}
+
+export default function MyComponent() {
+  return (
+    <Canvas>
+      <ambientLight intensity={0.8} />
+      <pointLight intensity={1} position={[0, 6, 0]} />
+      <MyScene />
+    </Canvas>
+  )
+}
+```
+
+## Working with spring events
```

Comment:
Maybe a good syncing example is syncing to an `Object3D` that is not represented in JSX. The `animated` examples wouldn't be able to do this and it doesn't require any complicated application design to demonstrate the strategy.

Reply:
> Maybe a good syncing example is syncing to an Object3D that is not represented in JSX.

Do you have an example that comes to mind? 🤔 I'm totally up for simplifying this.

Reply:
It mostly gets used for more complicated workflows where you want an `Object3D` proxy that doesn't have its matrix auto-updated because it is on the scene graph. For example, this is done in the Drei `Instance` component to calculate things like position before feeding it into the buffer array. I'm not sure offhand what the best condensed version of this would be, maybe something to do with instancing or a simple particle system where you are feeding the resulting matrices into a shader.

Reply:
That could be a great example to make. 
I think for the purpose of the doc i'll keep as is because its narrative is you're building up a thing over time.

Thanks for the thoughts!